In [23]:
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

**Dataset: IMDB - Principles**

In [37]:
prin = pd.read_csv('zippedData/imdb.title.principals.csv.gz')
prin.head(12)

tconst  ordering     nconst  category       job  \
0   tt0111414         1  nm0246005     actor       NaN   
1   tt0111414         2  nm0398271  director       NaN   
2   tt0111414         3  nm3739909  producer  producer   
3   tt0323808        10  nm0059247    editor       NaN   
4   tt0323808         1  nm3579312   actress       NaN   
5   tt0323808         2  nm2694680     actor       NaN   
6   tt0323808         3  nm0574615     actor       NaN   
7   tt0323808         4  nm0502652   actress       NaN   
8   tt0323808         5  nm0362736  director       NaN   
9   tt0323808         6  nm0811056  producer  producer   
10  tt0323808         7  nm0914939  producer  producer   
11  tt0323808         8  nm0779346  composer       NaN   

                  characters  
0                ["The Man"]  
1                        NaN  
2                        NaN  
3                        NaN  
4           ["Beth Boothby"]  
5          ["Steve Thomson"]  
6   ["Sir Lachlan Morrison"]  
7    ["Lady Delia Morrison"]  
8                        NaN  
9                        NaN  
10                       NaN  
11                       NaN

In [9]:
prin.ordering = prin.ordering.apply(lambda x: int(x))

In [14]:
prin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1028186 entries, 0 to 1028185
Data columns (total 6 columns):
tconst        1028186 non-null object
ordering      1028186 non-null int64
nconst        1028186 non-null object
category      1028186 non-null object
job           177684 non-null object
characters    393360 non-null object
dtypes: int64(1), object(5)
memory usage: 47.1+ MB


In [26]:
len(prin[(prin.category != prin.job) & prin.job.notna()].head())

5

Removing the job column only affects 5 rows (of <1 million)

In [29]:
prin.drop(columns='job', inplace = True)

In [31]:
len(prin[prin.category.isin(['actor', 'actress'])])

402926

At first glance we see characters column is more than half Null, but looking closer, we find that most actors have an assoiciated character. Even if 15% are Him/herself

In [75]:
prin.characters.value_counts().head()

Himself     43584
Herself     16127
Narrator     2218
Alex          656
David         621
Name: characters, dtype: int64

In [39]:
prin.characters =prin.characters.apply(lambda x: x.strip('["]') if isinstance(x, str) else x)

In [33]:
prin.ordering.unique()

array([ 1,  2,  3, 10,  4,  5,  6,  7,  8,  9], dtype=int64)

In [41]:
prin.ordering[prin.category =='director'].unique()

array([ 2,  5,  3,  6,  1,  4,  7, 10,  8,  9], dtype=int64)

-No connection between ordering and category

I'm tempted to convert the dataset to one line per film, but I think it suits our needs as it is

In [52]:
prin.to_csv('clean_data/clean_principles.csv', index = False)

**Dataset: IMDB - title basics**

In [76]:
bas = pd.read_csv('zippedData/imdb.title.basics.csv.gz')

In [124]:
bas.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 146143 entries, 0 to 146143
Data columns (total 6 columns):
tconst             146143 non-null object
primary_title      146143 non-null object
original_title     146143 non-null object
start_year         146143 non-null int64
runtime_minutes    146143 non-null float64
genres             140735 non-null object
dtypes: float64(1), int64(1), object(4)
memory usage: 12.8+ MB


In [79]:
bas.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [85]:
bas.primary_title.value_counts
bas.original_title.value_counts()
bas.start_year.value_counts()
bas.runtime_minutes.value_counts()

90.0     7131
80.0     3526
85.0     2915
100.0    2662
95.0     2549
         ... 
382.0       1
724.0       1
808.0       1
287.0       1
540.0       1
Name: runtime_minutes, Length: 367, dtype: int64

In [112]:
#fill null 'original_title' entries with the primary title
bas.loc[(bas.original_title.isna()),
        'original_title'] = bas.loc[(bas.original_title.isna()),
                                    'primary_title']

In [88]:
bas[bas.start_year == 2115]
#Checked it out, "upcoming experimental science fiction film" -year is correct

tconst primary_title original_title  start_year  runtime_minutes  \
89506  tt5174640     100 Years      100 Years        2115              NaN   

      genres  
89506  Drama

In [92]:
#But will remove as an extreme outlier; MS probably shan't want to replicate it
bas.drop(89506, inplace = True)

In [94]:
bas.describe()

start_year  runtime_minutes
count  146143.000000    114405.000000
mean     2014.621111        86.187247
std         2.720952       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2027.000000     51420.000000

In [120]:
#replacing null runtimes with the median
bas.loc[bas.runtime_minutes.isna(), 'runtime_minutes'] =87.0

In [121]:
bas.describe()
#Note: lowered the std, and interquartile range

start_year  runtime_minutes
count  146143.000000    146143.000000
mean     2014.621111        86.363753
std         2.720952       147.192154
min      2010.000000         1.000000
25%      2012.000000        75.000000
50%      2015.000000        87.000000
75%      2017.000000        95.000000
max      2027.000000     51420.000000

Genres are something I'll being looking at and only a small percentage are missing, so I'll drop rows

In [127]:
#create list of all genres
bas.dropna(inplace = True)

In [167]:
#clean list to remove multiples with only differing spaces
genres = []
for f in bas.genres:
    for g in f.split(','):
        if not g in genres:
            genres.append(g)

In [182]:
#create a column for each genre, give value True if the string appears 
#...in the original genre_ids column
for g in genres:
    bas[g] = bas.genres.apply(lambda x: g in x)

In [194]:
bas[['genres']+ genres].head()

genres  Action  Crime  Drama  Biography  Comedy  Fantasy  \
0    Action,Crime,Drama    True   True   True      False   False    False   
1       Biography,Drama   False  False   True       True   False    False   
2                 Drama   False  False   True      False   False    False   
3          Comedy,Drama   False  False   True      False    True    False   
4  Comedy,Drama,Fantasy   False  False   True      False    True     True   

   Horror  Thriller  Adventure  ...  Music  Sport  Western  Musical  Adult  \
0   False     False      False  ...  False  False    False    False  False   
1   False     False      False  ...  False  False    False    False  False   
2   False     False      False  ...  False  False    False    False  False   
3   False     False      False  ...  False  False    False    False  False   
4   False     False      False  ...  False  False    False    False  False   

    News  Talk-Show  Reality-TV  Game-Show  Short  
0  False      False       False      False  False  
1  False      False       False      False  False  
2  False      False       False      False  False  
3  False      False       False      False  False  
4  False      False       False      False  False  

[5 rows x 28 columns]

In [195]:
bas.to_csv('clean_data/clean_basics.csv', index = False)

**Dataset: IMDB - name**

In [131]:
nam = pd.read_csv('zippedData/imdb.name.basics.csv.gz')

In [132]:
nam.head()

nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256

In [134]:
nam.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 606648 entries, 0 to 606647
Data columns (total 6 columns):
nconst                606648 non-null object
primary_name          606648 non-null object
birth_year            82736 non-null float64
death_year            6783 non-null float64
primary_profession    555308 non-null object
known_for_titles      576444 non-null object
dtypes: float64(2), object(4)
memory usage: 27.8+ MB


The main use we'll have for this dataset is just to attach names to other tables via nconst

In [135]:
nam.describe()

birth_year   death_year
count  82736.000000  6783.000000
mean    1967.043826  2000.523367
std       22.122190    43.951530
min        1.000000    17.000000
25%     1957.000000  2001.000000
50%     1971.000000  2013.000000
75%     1981.000000  2016.000000
max     2014.000000  2019.000000

In [151]:
nam[nam.birth_year < 1650]

nconst                    primary_name  birth_year  death_year  \
1244    nm0963344                     John Bunyan      1628.0      1688.0   
11901   nm0549265             Christopher Marlowe      1564.0      1593.0   
21094   nm0613556                Shikibu Murasaki       973.0      1031.0   
38106   nm0674518                Charles Perrault      1628.0      1703.0   
39848   nm0596942                         Molière      1622.0      1673.0   
52088   nm1063158                     Cheng'en Wu      1506.0      1581.0   
52523   nm0148859  Miguel de Cervantes y Saavedra      1547.0      1616.0   
59713   nm0000636             William Shakespeare      1564.0      1616.0   
71418   nm0478548            Madame de La Fayette      1634.0      1693.0   
74627   nm0017266              Mariana Alcoforado      1640.0      1723.0   
80184   nm0090504              Giovanni Boccaccio      1313.0      1375.0   
82989   nm0699700                     Songling Pu      1640.0      1715.0   
112096  nm0019604                 Dante Alighieri      1265.0      1321.0   
135957  nm2228277             Giambattista Basile      1566.0      1632.0   

                     primary_profession  \
1244                  writer,soundtrack   
11901                 writer,soundtrack   
21094                            writer   
38106                            writer   
39848   writer,soundtrack,miscellaneous   
52088                            writer   
52523                 writer,soundtrack   
59713   writer,soundtrack,miscellaneous   
71418                            writer   
74627                            writer   
80184                            writer   
82989                            writer   
112096                writer,soundtrack   
135957                           writer   

                               known_for_titles  
1244    tt0393584,tt0194913,tt0234464,tt1336619  
11901   tt0114279,tt0062898,tt0287837,tt0101798  
21094   tt0092875,tt0043580,tt0094100,tt0256385  
38106   tt0053285,tt1587310,tt0042332,tt8651654  
39848   tt0154249,tt0026069,tt0017448,tt0370853  
52088   tt1163129,tt6168860,tt0112778,tt1753783  
52523   tt0204285,tt6151214,tt0023956,tt0039330  
59713   tt8632012,tt3894536,tt5932378,tt7344870  
71418   tt1599975,tt0158117,tt0204761,tt1263778  
74627             tt0062836,tt0076312,tt2219674  
80184   tt0045675,tt0068832,tt0158510,tt5666304  
82989   tt0101754,tt0059029,tt1625192,tt0064451  
112096  tt1874782,tt7370578,tt1598441,tt1118686  
135957  tt0177448,tt1212157,tt3278330,tt0061433

As I go back looking for implausible birth years, the more I find they are correct (e.g. Murasaki Shikibu (紫 式部, English: Lady Murasaki; c. 973 or 978 – c. 1014 or 1031) was a Japanese novelist, poet and lady-in-waiting at the Imperial court during the Heian period. -Wikipedia). Except for Michael Vignola, born (Nov 1st) 1980

In [150]:
nam.loc[nam['birth_year'] == 1.0, 'birth_year'] = 1980.0

In [152]:
nam.death_year.value_counts()

2018.0    624
2017.0    621
2016.0    592
2015.0    540
2013.0    437
         ... 
1847.0      1
1848.0      1
1850.0      1
1854.0      1
1031.0      1
Name: death_year, Length: 214, dtype: int64

Despite only a small fraction having birth years, it's still a large enough number if to do something separate with, so I'll leave it as is. However I don't think there's enough data in the death_year column to be of use.

In [160]:
nam.drop(columns = 'death_year', inplace = True)

For primary_profession and known_for_titles, I'll leave the null values as they are. Too many entries to drop the column, not worth dropping the rows for a column that might not be used.

In [163]:
nam.primary_profession.value_counts()
nam.known_for_titles.value_counts()

In [164]:
nam.to_csv('clean_data/clean_names.csv', index = False)

**Dataset: TMDB -Movies**

In [3]:
tmdb = pd.read_csv('zippedData/tmdb.movies.csv.gz')
tmdb.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [4]:
tmdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
Unnamed: 0           26517 non-null int64
genre_ids            26517 non-null object
id                   26517 non-null int64
original_language    26517 non-null object
original_title       26517 non-null object
popularity           26517 non-null float64
release_date         26517 non-null object
title                26517 non-null object
vote_average         26517 non-null float64
vote_count           26517 non-null int64
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


In [5]:
tmdb.genre_ids.value_counts()

[99]                 3700
[]                   2479
[18]                 2268
[35]                 1660
[27]                 1145
                     ... 
[27, 80, 18]            1
[10751, 80, 18]         1
[878, 80]               1
[10402, 14, 35]         1
[9648, 18, 10770]       1
Name: genre_ids, Length: 2477, dtype: int64

In [6]:
genre_dict = {'28':"Action", '12':"Adventure", '16':"Animation", '35':"Comedy", 
              '80':"Crime", '99':"Documentary", '18':"Drama", '10751':"Family", 
              '14':"Fantasy", '36':"History", '27':"Horror", '10402':"Music", 
              '9648':"Mystery", '10749':"Romance", '878':"Science Fiction", 
              '10770':"TV Movie", '53':"Thriller", '10752':"War",'37':"Western"}

In [7]:
#create list of all genres
genres = []
for f in tmdb.genre_ids:
    for g in f.strip('[]').split(','):
        if not g in genres:
            genres.append(g)

In [8]:
#clean list to remove multiples with only differing spaces
genres = set(list(map(lambda x: x.strip(' '), genres)))

In [9]:
#create a column for each genre, give value True if the string appears 
#...in the original genre_ids column
for g in genres:
    tmdb[g] = tmdb.genre_ids.apply(lambda x: g in x)

In [10]:
tmdb.head(2)

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   

                                          title  vote_average  vote_count  \
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788   
1                      How to Train Your Dragon           7.7        7610   

            35     80     37     53     28     18  10402  10751     36     27  \
0  True  False  False  False  False  False  False  False   True  False  False   
1  True  False  False  False  False  False  False  False   True  False  False   

      99  10752    14    878  10770     16    12  10749   9648  
0  False  False  True  False  False  False  True  False  False  
1  False  False  True  False  False   True  True  False  False

In [11]:
tmdb.drop(columns ='', inplace = True)

In [12]:
tmdb.rename(columns = genre_dict, inplace = True)

In [13]:
tmdb.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  \
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788   
1                      How to Train Your Dragon           7.7        7610   
2                                    Iron Man 2           6.8       12368   
3                                     Toy Story           7.9       10174   
4                                     Inception           8.3       22186   

   Comedy  Crime  Western  Thriller  Action  Drama  Music  Family  History  \
0   False  False    False     False   False  False  False    True    False   
1   False  False    False     False   False  False  False    True    False   
2   False  False    False     False    True  False  False   False    False   
3    True  False    False     False   False  False  False    True    False   
4   False  False    False     False    True  False  False   False    False   

   Horror  Documentary    War  Fantasy  Science Fiction  TV Movie  Animation  \
0   False        False  False     True            False     False      False   
1   False        False  False     True            False     False       True   
2   False        False  False    False             True     False      False   
3   False        False  False    False            False     False       True   
4   False        False  False    False             True     False      False   

   Adventure  Romance  Mystery  
0       True    False    False  
1       True    False    False  
2       True    False    False  
3      False    False    False  
4       True    False    False

In [14]:
tmdb.iloc[:,:10]
tmdb.iloc[:,10:].sort_index(axis=1)  #genre columns in alphabetical order
tmdb = tmdb.iloc[:,:10].join(tmdb.iloc[:,10:].sort_index(axis=1))

In [15]:
tmdb.drop(columns = ['Unnamed: 0', 'genre_ids'], inplace = True)

In [16]:
tmdb.id.value_counts()

292086    3
463839    3
11976     3
391872    3
416572    3
         ..
356987    1
350846    1
479871    1
500353    1
524288    1
Name: id, Length: 25497, dtype: int64

In [17]:
tmdb[tmdb.id == 292086] #duplicate entries found

id original_language original_title  popularity release_date  \
5838   292086                en   Come Morning       3.013   2012-10-21   
11836  292086                en   Come Morning       3.013   2012-10-21   
21565  292086                en   Come Morning       3.013   2012-10-21   

              title  vote_average  vote_count  Action  Adventure  Animation  \
5838   Come Morning           6.3           5   False      False      False   
11836  Come Morning           6.3           5   False      False      False   
21565  Come Morning           6.3           5   False      False      False   

       Comedy  Crime  Documentary  Drama  Family  Fantasy  History  Horror  \
5838    False  False        False   True   False    False    False   False   
11836   False  False        False   True   False    False    False   False   
21565   False  False        False   True   False    False    False   False   

       Music  Mystery  Romance  Science Fiction  TV Movie  Thriller    War  \
5838   False    False    False            False     False      True  False   
11836  False    False    False            False     False      True  False   
21565  False    False    False            False     False      True  False   

       Western  
5838     False  
11836    False  
21565    False

In [18]:
tmdb.drop_duplicates(inplace = True)

In [34]:
#check for common implausible values
tmdb.original_language.value_counts()
tmdb.original_title.value_counts()
tmdb.popularity.value_counts()
tmdb.release_date.value_counts()
tmdb.title.value_counts()
tmdb.vote_average.value_counts()
tmdb.vote_count.value_counts()

1       6438
2       2994
3       1720
4       1303
5        945
        ... 
953        1
969        1
985        1
1001       1
2049       1
Name: vote_count, Length: 1693, dtype: int64

In [41]:
tmdb.to_csv('clean_data/clean_tmdb.cs', index = False)

In [19]:
tmdb.describe()

id    popularity  vote_average    vote_count
count   25497.000000  25497.000000  25497.000000  25497.000000
mean   294203.960505      3.043279      5.979331    178.795780
std    154690.249660      4.261045      1.866094    914.150311
min        27.000000      0.600000      0.000000      1.000000
25%    154770.000000      0.600000      5.000000      1.000000
50%    307125.000000      1.321000      6.000000      5.000000
75%    420707.000000      3.490000      7.000000     25.000000
max    608444.000000     80.773000     10.000000  22186.000000

In [21]:
tmdb.sort_values(by=['vote_count']).tail(20)

id original_language                      original_title  \
14172  140607                en        Star Wars: The Force Awakens   
14184  150540                en                          Inside Out   
2468    10195                en                                Thor   
2474     1771                en  Captain America: The First Avenger   
14169   99861                en             Avengers: Age of Ultron   
17437  297761                en                       Suicide Squad   
5182    49026                en               The Dark Knight Rises   
23811  299536                en              Avengers: Infinity War   
17382  271110                en          Captain America: Civil War   
14193  135397                en                      Jurassic World   
7881    68721                en                          Iron Man 3   
14177   76341                en                  Mad Max: Fury Road   
5235    70160                en                    The Hunger Games   
5189    68718                en                    Django Unchained   
11021  118340                en             Guardians of the Galaxy   
11032  157336                en                        Interstellar   
6       19995                en                              Avatar   
5179    24428                en                        The Avengers   
17383  293660                en                            Deadpool   
4       27205                en                           Inception   

       popularity release_date                               title  \
14172      32.281   2015-12-18        Star Wars: The Force Awakens   
14184      24.797   2015-06-19                          Inside Out   
2468       38.068   2011-05-06                                Thor   
2474       25.808   2011-07-22  Captain America: The First Avenger   
14169      44.383   2015-05-01             Avengers: Age of Ultron   
17437      16.530   2016-08-05                       Suicide Squad   
5182       26.220   2012-07-20               The Dark Knight Rises   
23811      80.773   2018-04-27              Avengers: Infinity War   
17382      39.137   2016-05-06          Captain America: Civil War   
14193      20.709   2015-06-12                      Jurassic World   
7881       32.093   2013-05-03                          Iron Man 3   
14177      28.099   2015-05-15                  Mad Max: Fury Road   
5235       14.212   2012-03-23                    The Hunger Games   
5189       21.260   2012-12-25                    Django Unchained   
11021      49.606   2014-08-01             Guardians of the Galaxy   
11032      28.440   2014-11-05                        Interstellar   
6          26.526   2009-12-18                              Avatar   
5179       50.289   2012-05-04                        The Avengers   
17383      35.067   2016-02-12                            Deadpool   
4          27.920   2010-07-16                           Inception   

       vote_average  vote_count  Action  Adventure  Animation  Comedy  Crime  \
14172           7.4       12641    True       True      False   False  False   
14184           8.0       12691   False      False       True    True  False   
2468            6.7       12764    True       True      False   False  False   
2474            6.9       12810    True       True      False   False  False   
14169           7.3       13457    True       True      False   False  False   
17437           5.9       13533    True       True      False   False   True   
5182            7.7       13933    True      False      False   False   True   
23811           8.3       13948    True       True      False   False  False   
17382           7.4       14000    True       True      False   False  False   
14193           6.6       14056    True       True      False   False  False   
7881            6.9       14155    True       True      False   False  False   
14177           7.4       14454    True       True      False   False  False   
5235    

In [25]:
tmdb[tmdb['Western']].sort_values(by = ['release_date'])

id original_language  \
525     71244                en   
2434    79540                en   
1634   114219                en   
2202   248412                en   
862     29776                en   
1445    39049                en   
884     40852                en   
1150   175987                en   
836    101520                en   
53      20533                en   
1665    41361                en   
404     41988                en   
882     43700                en   
905     39807                en   
742     46562                en   
290     57120                en   
639     77594                en   
364     45569                en   
1046    49594                en   
222     46528                en   
59      44264                en   
4158   100095                en   
3762   111166                en   
3683   268956                en   
3471    69396                en   
3454    55093                it   
3840    91322                en   
4725    61794                en   
2520    44896                en   
3435    98549                en   
4977    78243                en   
4078    57817                en   
3796    70197                en   
3246    71548                en   
2497    49849                en   
4368   453454                en   
4838    92127                en   
3567    67328                en   
3498   123969                en   
3420    87311                en   
2780    68818                en   
3853   146946                en   
7757   109440                en   
6848   282675                en   
7702   115033                en   
7011    89828                en   
5979    89888                en   
5415    80304                en   
6065   147764                en   
6884   107033                fr   
5929   109500                en   
6637   109415                en   
5925   117700                en   
6551   116340                en   
7469   134693                en   
7479   111949                en   
6558   252328                en   
6309    83007                en   
5189    68718                en   
8242   158916                en   
9115   170776                en   
9776   278220                en   
10881  297588                en   
8948   185457                fr   
7967    57201                en   
9434   238822                en   
8105   152748                en   
10764  267924                es   
10965  217331                en   
10921  232517                en   
10762  293574                en   
8112   151933                en   
9530    92780                es   
12915  252660                en   
12103  249616                en   
12747  267288                en   
11923  278878                en   
8735   169800                en   
11154  188161                en   
12923  272724                en   
11966  238811                zh   
12545  326045                en   
11286  264656                en   
13239  599687                en   
11189  259611                en   
11436  284284                en   
11525  215379                en   
11139  266285                da   
16832  368717                en   
16622  335004                en   
16240  354100                en   
14332  223485                en   
15327  333594                en   
16826  379092                en   
16262  344494                en   
15268  267579                es   
14391  362046                en   
16134  356849                en   
17038  319093                en   
15416  280005                en   
14278  294963                en   
14670  421281                en   
16666  500986                en   
17352  370458                en   
14313  347969                en   
14203  281957                en   
14194  273248                en   
19095  377357                en   
14467  362703                en   
17733  376501                en   
17654  174751                en   
17611  354110                en   
18709  385309                en   
18399  384450        